pkg/process/metadata/parser/java/util.go

```diff
@@ -0,0 +1,126 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package javaparser
+
+import (
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/rickar/props"
+	"github.com/vibrantbyte/go-antpath/antpath"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+type mapSource struct {
```

Comment:
what's the purpose of the struct?

Reply:
I added some explain in the comment below and some comments as well to the struct